lib/mixlib/install/backend/package_router.rb

```diff
@@ -0,0 +1,222 @@
+
+          # Merge properties to a flat Hash
+          results.collect! do |result|
+            {
+              "filename" => result["name"],
+            }.merge(
+              map_properties(result["properties"])
+            )
+          end
+
+          # Convert results to build records
+          results.map { |a| create_artifact(a) }
+        end
+
+        #
+        # GET request
+        #
+        def get(url)
+          uri = URI.parse(endpoint)
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.use_ssl = (uri.scheme == "https")
+
+          full_path = File.join(uri.path, url)
+          request = Net::HTTP::Get.new(full_path)
+
+          res = http.request(request)
+
+          body = if res.code != "200"
+                   '{"results":[]}'
+                 else
+                   res.body
+                 end
+          JSON.parse(body)
+        end
+
+        def create_artifact(artifact_map)
+          platform, platform_version = normalize_platform(artifact_map["omnibus.platform"],
+            artifact_map["omnibus.platform_version"])
+
+          chef_standard_path = generate_chef_standard_path(options.channel,
+            artifact_map["omnibus.project"],
+            artifact_map["omnibus.version"],
+            platform,
+            platform_version,
+            artifact_map["filename"]
+          )
+
+          ArtifactInfo.new(
+            md5:              artifact_map["omnibus.md5"],
+            sha256:           artifact_map["omnibus.sha256"],
+            sha1:             artifact_map["omnibus.sha1"],
+            version:          artifact_map["omnibus.version"],
+            platform:         platform,
+            platform_version: platform_version,
+            architecture:     normalize_architecture(artifact_map["omnibus.architecture"]),
+            url:              chef_standard_path
+          )
+        end
+
+        private
+
+        # Converts Array<Hash> where the Hash is a key pair and
+        # value pair to a simplifed key/pair Hash
+        #
+        def map_properties(properties)
+          return {} if properties.nil?
+          properties.each_with_object({}) do |prop, h|
+            h[prop["key"]] = prop["value"]
+          end
+        end
+
+        # Generates a chef standard download uri in the form of
+        # http://endpoint/files/:channel/:project/:version/:platform/:platform_version/:file
+        def generate_chef_standard_path(channel, project, version, platform, platform_version, filename)
+          uri = []
+          uri << endpoint.sub(/\/$/, "")
+          uri << "files"
+          uri << channel
+          uri << project
+          uri << version
+          uri << platform
+          uri << platform_version
+          uri << filename
+          uri.join("/")
+        end
+
+        def endpoint
+          @endpoint ||= ENV.fetch("PACKAGE_ROUTER_ENDPOINT", ENDPOINT)
```

Comment:
We'll need to port some of the old `Mixlib::Install::Backend::Bintray#url` [method](https://github.com/chef/mixlib-install/blob/d04dc39b68847f654d51b7ff1f97fa5806b9b477/lib/mixlib/install/backend/bintray.rb#L166-L171) into this class somewhere, namely the bit that used a non-SNI, non-SSL version for legacy platforms:

```ruby
base_url = case "#{platform_info[:platform]}-#{platform_info[:platform_version]}"
           when "freebsd-9", "el-5", "solaris2-5.9", "solaris2-5.10"
             COMPAT_DOWNLOAD_URL_ENDPOINT
           else
             DOWNLOAD_URL_ENDPOINT
           end
```

The new `COMPAT_DOWNLOAD_URL_ENDPOINT` should be `http://packages.chef.io`.